roles/uperf/templates/service.yml.j2

```
@@ -1,27 +1,43 @@
 ---
-kind: Service
 apiVersion: v1
-metadata:
-  name: uperf-service-{{ item }}-{{ trunc_uuid }}
-  namespace: '{{ operator_namespace }}'
-  labels:
-    app: uperf-bench-server-{{ item }}-{{ trunc_uuid }}
-    type: uperf-bench-server-{{ trunc_uuid }}
-spec:
-  selector:
-    app: uperf-bench-server-{{ item }}-{{ trunc_uuid }}
-  ports:
-  - name: uperf
-    port: 20000
-    targetPort: 20000
-    protocol: TCP
+kind: List
+metadata: {}
+items:
+{% macro job_template(item, node_idx_item='') %}
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name:  uperf-service-{{worker_node_list[ node_idx_item ] | truncate(32,true,'') }}-{{ item }}-{{ trunc_uuid }}
+
+      namespace: '{{ operator_namespace }}'
+      labels:
+        app:  uperf-bench-server-{{worker_node_list[ node_idx_item ] | truncate(32,true,'') }}-{{ item }}-{{ trunc_uuid }}
+        type: uperf-bench-server-{{ trunc_uuid }}
+      annotations:
+            node_idx: '{{ node_idx_item }}'
+            pod_idx: '{{ item }}'
+    spec:
+      selector:
+        app:  uperf-bench-server-{{worker_node_list[ node_idx_item ] | truncate(32,true,'')}}-{{ item }}-{{ trunc_uuid }}
```

The service selector also needs to be updated.

```suggestion
        app: uperf-bench-server-{{ item  }}-{{ trunc_uuid }}
```